pywhy_stats/kci.py

```diff
@@ -0,0 +1,196 @@
+from typing import Callable
+
+import numpy as np
+from numpy import shape, ndarray
+from numpy.linalg import eigvalsh
+from numpy.random import permutation
+from scipy import stats
+
+from pywhy_stats.custom_kernels import rbf_kernel
+from pywhy_stats.pvalue_result import PValueResult
+
+
+def ind(X: np.ndarray,
+        Y: np.ndarray,
+        kernel_X: Callable[[np.ndarray], np.ndarray] = rbf_kernel,
+        kernel_Y: Callable[[np.ndarray], np.ndarray] = rbf_kernel,
+        approx: bool = True,
+        null_ss: int = 1000) -> PValueResult:
```

Comment:
Should we also expose any keyword arguments to the kernel function? e.g. `**kernel_X_kwargs` and ``*kernel_Y_kwargs``, or `kwidth_X`, etc.

Not sure, so just asking here...

Reply:
No, the idea is that, if a user wants to customize the kernel calls, they can just provide it via partial. E.g., 
```
kci.ind(X, Y, kernel_X = partial(rbf_kernel, width=1.4, my_other_rbf_parameter=...)
```
This way, we avoid keeping track of all kind of parameters for all kind of kernels.

Reply:
Okay. I think I'm on board. It would be great to also have this pattern as a MWE in the `Examples` docstring. Then users can easily copy-paste the pattern.